Fix problem with disabled SSH password login

pdostal opened the pull request:
As this problem appeared on Tumbleweed where the new directory
`/etc/ssh/sshd_config.d/` is in place I'm checking for it.

- Related ticket: [poo#93850](https://progress.opensuse.org/issues/93850)
- Verification run: [openVPN](http://pdostal-server.suse.cz/tests/12211) [Wireguard](http://pdostal-server.suse.cz/tests/12215)

DimStar77 commented:
since c6d49182c27a48bc23e5989e07c0dccb200ee0dc we have a `permit_root_ssh` function in lib/utils.pm

Any chance to reuse that one here as well?

rfan1 commented:
> since [c6d4918](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c6d49182c27a48bc23e5989e07c0dccb200ee0dc) we have a `permit_root_ssh` function in lib/utils.pm
> 
> Any chance to reuse that one here as well?

Yes, https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12697/

DimStar77 commented:
> > since [c6d4918](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c6d49182c27a48bc23e5989e07c0dccb200ee0dc) we have a `permit_root_ssh` function in lib/utils.pm
> > Any chance to reuse that one here as well?
> 
> Yes, #12697

that was the MR that merged the commit I referenced - but that commit does not fix the multimachine setups for tests like wireguard - so I'm not really sure I understand your comment

rfan1 commented:
> > > since [c6d4918](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c6d49182c27a48bc23e5989e07c0dccb200ee0dc) we have a `permit_root_ssh` function in lib/utils.pm
> > > Any chance to reuse that one here as well?
> > 
> > 
> > Yes, #12697
> 
> that was the MR that merged the commit I referenced - but that commit does not fix the multimachine setups for tests like wireguard - so I'm not really sure I understand your comment

Just adding my PR for my commit "permit_root_ssh" function, hopefully it can help.

pdostal commented:
Thank you, @DimStar77, @rfan1 ! :) I will update my PR.

DimStar77 commented on tests/network/setup_multimachine.pm:
```diff
@@ -16,8 +16,8 @@ use warnings;
 use mm_network 'setup_static_mm_network';
-use utils 'zypper_call';
-use Utils::Systemd 'disable_and_stop_service';
+use utils 'zypper_call permit_root_ssh';
```
`use utils qw(zypper_call permit_root_ssh)`

DimStar77 commented:
Thanks for the updated PR - can you please ensure there is an up-to-date verification run linked?

pdostal commented:
@DimStar77 foxed, reverified. Thank you!

DimStar77 commented:
AS this is a multi-machine test: can you please verify what happened to the linked/dependent tests?

http://pdostal-server.suse.cz/tests/12190#dependencies -> client failed

pdostal commented:
> AS this is a multi-machine test: can you please verify what happened to the linked/dependent tests?
> 
> http://pdostal-server.suse.cz/tests/12190#dependencies -> client failed

that is unrelated to the issue. I added fixed openVPN run and added Wireguard run.
